JetStreamDriver.js

```diff
@@ -1198,58 +1205,42 @@ class WasmEMCCBenchmark extends AsyncBenchmark {
                 },
             };
             globalObject.Module = Module;
-            `;
-        return str;
+        `;
     }
 
-    // FIXME: Why is this part of the runnerCode and not prerunCode?
+    // This is in runnerCode rather than prerunCode because prerunCode isn't currently structured to be async by default.
     get runnerCode() {
-        let str = `function loadBlob(key, path, andThen) {`;
-
+        let str = `(async function doRunWrapper() {`
         if (isInBrowser) {
             str += `
-                var xhr = new XMLHttpRequest();
-                xhr.open('GET', path, true);
-                xhr.responseType = 'arraybuffer';
-                xhr.onload = function() {
-                    Module[key] = new Int8Array(xhr.response);
-                    andThen();
-                };
-                xhr.send(null);
+                async function getBinary(key, blobURL) {
+                    const response = await fetch(blobURL);
+                    Module[key] = new Int8Array(await response.arrayBuffer());
+                }
             `;
-        } else {
+        } else
             str += `
-            Module[key] = new Int8Array(read(path, "binary"));
-
-            Module.setStatus = null;
-            Module.monitorRunDependencies = null;
-
-            Promise.resolve(42).then(() => {
-                try {
-                    andThen();
-                } catch(e) {
-                    console.log("error running wasm:", e);
-                    console.log(e.stack);
-                    throw e;
+                // Needed because SpiderMonkey shell doesn't have a setTimeout.
+                Module.setStatus = null;
+                Module.monitorRunDependencies = null;
+                function getBinary(key, path) {
+                    Module[key] = new Int8Array(read(path, "binary"));
                 }
-            })
             `;
-        }
-
-        str += "}";
 
-        let keys = Object.keys(this.plan.preload);
-        for (let i = 0; i < keys.length; ++i) {
-            str += `loadBlob("${keys[i]}", "${this.plan.preload[keys[i]]}", async () => {\n`;
+        for (let [ preloadKey, blobURLOrPath ] of this.preloads) {
+            if (preloadKey == "wasmBinary") {
```

Comment:
Can we get rid of this special casing of just the `wasmBinary` preloadKey? There are not that many Emscripten workloads, so I'd be happy to just add a line `Module["wasmBinary"] = genericPreloadFunc(wasmBinary)` to every one of those (for a little less "magic").